### PR TITLE
Feature/preprocessing

### PR DESCRIPTION
**Pull request by Inderdeep-Singh02**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Mind-Matrix: LLM Powered Data Synergy
```
Comment (Hirkani2003) [inline README.md:1]: Add description about the project

File: application/main/routers/preprocess.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from application.main.utils.preprocessing.regression.regresion import reg_preprocessing 
+
+
+
+
+router = APIRouter(prefix='/preprocess')
+
+@router.get('/')
+def pp():
+    return JSONResponse(content={"mmm":"jjj"}, status_code=200)
```
Comment (Hirkani2003) [inline application/main/routers/preprocess.py]: remove this

File: application/main/routers/preprocess.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+@router.post('/')
+async def preprocess(request: Request, csvFile: UploadFile = File(...)):
+    
+    try:
+        df = await dataframe(csvFile=csvFile)
+        preprocessed_df = reg_preprocessing(df, target=-1)
```
Comment (Hirkani2003) [inline application/main/routers/preprocess.py]: give better name to the function

File: application/main/utils/preprocessing/regression/regresion.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+from sklearn.preprocessing import LabelEncoder, StandardScaler
+from application.main.utils.preprocessing.outlier import remove_outliers
+
+def reg_preprocessing(df, target=None):
```
Comment (Hirkani2003) [inline application/main/utils/preprocessing/regression/regresion.py]: write comments about all the procedure steps

File: application/main/utils/preprocessing/regression/regresion.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        target_column = target
+        target_index = df.columns.get_loc(target_column)
+    elif isinstance(target, int):
+        target_index = target
+        target_column = df.columns[target_index]
+    else:
+        raise ValueError("Invalid target specification. Provide either column name or index.")
+
+
+    numeric_columns = df.select_dtypes(exclude=['object']).columns.difference(label_encoded_columns)
```
Comment (Hirkani2003) [inline application/main/utils/preprocessing/regression/regresion.py]: Exclude target column from standardization and normalization.

File: docker-compose.yaml [hunk 6]
```diff
@@ -2,7 +2,7 @@ version: '3.8'
 
 services:
   api:
-    image: isadeep/mindmatrix:v1.0
+    image: isadeep/mindmatrix:v1.1
```
Comment (Hirkani2003) [inline docker-compose.yaml]: build locally and take latest changes.
